### ISSUE TITLE
Incorrect installation when using pipenv

### ISSUE THREAD

**Issue by rodmaz** [uber/h3-py#33]
When user attempts to install `h3` using `pipenv`, no `/out` directory is generated.

This does **not** work:
`pipenv install h3`

This does work:

`pipenv run pip install h3`

Any idea why the behaviour is different? This is a problem in production environments, whose deployment is performed with `pipenv`.

**Comment by karentycoon** [comment]
Thank you for reporting this problem! 
@rodmaz Are you running in OSX or Linux in production environments? Could you try remove following cache dir before install?
~/Library/Caches/pipenv
~/Library/Caches/pip
~/Library/Caches/pip-tools?

I am able to install using pipenv in my laptop. Do you see anything different from the below logs?
`< pipenv install h3
Creating a virtualenv for this project…
Pipfile: /Users/taikun/Documents/projects/h3-py/Pipfile
Using /usr/local/Cellar/pipenv/2018.11.26_2/libexec/bin/python3.7 (3.7.2) to create virtualenv…
⠼ Creating virtual environment...Already using interpreter /usr/local/Cellar/pipenv/2018.11.26_2/libexec/bin/python3.7
Using real prefix '/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7'
New python executable in /Users/taikun/.virtualenvs/h3-py-98KwwbZd/bin/python3.7
Also creating executable in /Users/taikun/.virtualenvs/h3-py-98KwwbZd/bin/python
Installing setuptools, pip, wheel...
done.

✔ Successfully created virtual environment!
Virtualenv location: /Users/taikun/.virtualenvs/h3-py-98KwwbZd
Creating a Pipfile for this project…
Installing h3…
Adding h3 to Pipfile's [packages]…
✔ Installation Succeeded
Pipfile.lock not found, creating…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
✔ Success!
Updated Pipfile.lock (b21aa3)!
Installing dependencies from Pipfile.lock (b21aa3)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 1/1 — 00:00:00
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
17:34:18 (master) taikun@taikun-C02RT0QJG8WM:~/Documents/projects/h3-py
< pipenv shell
Launching subshell in virtual environment…
 . /Users/taikun/.virtualenvs/h3-py-98KwwbZd/bin/activate
17:34:25 (master) taikun@taikun-C02RT0QJG8WM:~/Documents/projects/h3-py
<  . /Users/taikun/.virtualenvs/h3-py-98KwwbZd/bin/activate
(h3-py) 17:34:25 (master) taikun@taikun-C02RT0QJG8WM:~/Documents/projects/h3-py
< pip freeze | grep h3
h3==3.4.2
(h3-py) 17:34:34 (master) taikun@taikun-C02RT0QJG8WM:~/Documents/projects/h3-py
< python
Python 3.7.2 (default, Mar 12 2019, 17:17:03)
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from h3 import h3
>>> exit();`

**Comment by rodmaz** [comment]
Thanks @karentycoon for your prompt reply. I am using macOS (Mojave). Production uses Linux.
Indeed when I cleaned up the cache folders, it worked. Do you have an idea why?
I will keep an eye because it seems I am not the only one having this problem.

**Comment by dfellis** [comment]
The issue is that the [`setup.py` file doesn't error out when `.install.sh` returns a non-zero status](https://github.com/uber/h3-py/blob/master/setup.py#L15) (it doesn't check the return code, basically) so `pip` **thinks** it was built successfully and caches it even if it errors because it's missing a build tool or the git clone fails or whatever. Then subsequent reinstalls just use the broken cached version instead of actually rebuilding it.

**Comment by rodmaz** [comment]
We are deploying the library on a AWS EMR (Hadoop-Spark) cluster.  Well apparently it does not work either when we install it on Amazon Linux.

```
[ec2-user@ip-xxx-xxx-xxx-xxx site-packages]$ cd h3
[ec2-user@ip-xxx-xxx-xxx-xxx h3]$ ls -la
total 44
drwxr-xr-x  3 root root  4096 Mar 13 21:59 .
drwxr-xr-x 21 root root  4096 Mar 13 22:02 ..
-rw-r--r--  1 root root 25103 Mar  5 00:05 h3.py
-rw-r--r--  1 root root    25 Mar  1 22:09 __init__.py
drwxr-xr-x  2 root root  4096 Mar 13 21:59 __pycache__
[ec2-user@ip-xxx-xxx-xxx-xxx]$ pwd
/usr/local/lib64/python3.6/site-packages/h3
```

This is totally unreliable.

**Comment by rodmaz** [comment]
I think this ticket can be safely closed. 
However it would be much better if the installation clearly failed when an error occurred and not showed as "successful" even though it was not (and worst, stayed corrupt in cache).

**Comment by karentycoon** [comment]
@rodmaz https://github.com/uber/h3-py/pull/34 this should fix the problem you described. 
We will close this issue after the above fix is applied in the new version.

**Comment by karentycoon** [comment]
Thanks for reporting! We just released new version 3.4.3 and this issue is fixed in the new version.
